programmers/난이도별/level01.소수_찾기/Leemainsw.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @param num: number 숫자
+ * @returns number
+ */
+function solution(num: number): number {
+    // 1 ~ num 길이 만큼의 배열을 생성해서 1을 채운다.
+    const arr: Array<number> = new Array(num).fill(1);
+    
+    // 1은 소수가 아니기 때문에 0으로 변경
+    arr[0] = 0;
+    
+    // 2부터 자신의 제곱근을 돌면서 자신의 배수를 0으로 바꾼다.
+    for(let i: number = 2; i*i <= num; i++){ //2,3,4,5,
+        for(let j:number = i*i; j<=num; j+=i){ // (2,4,8..), (3,6,9..)
```

Comment:
```suggestion
    for(let i = 2; i*i <= num; i++){ //2,3,4,5,
        for(let j = i*i; j<=num; j+=i){ // (2,4,8..), (3,6,9..)
```
이런 경우에는 타입 선언 해주지 않아도 괜찮습니다! `= 2`를 사용하면 `number`로 타입 추론이 되고 `undefined | null` 이 발생하지 않기 때문입니다.

https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html
공식 도큐먼트의 가장 아랫부분을 보아도 `:number`는 설정해주지 않고 있어요.

Reply:
확실히 이곳에서는 number만 사용되니 굳이 타입 설정을 하지 않는게 더 효율적인 코드였을 것 같아요-! 💡